packages/astro/src/vite-plugin-markdown/index.ts

```diff
@@ -151,7 +155,7 @@ ${setup}`.trim();
 
 				// Transform from `.astro` to valid `.ts`
 				let { code: tsResult } = await transform(astroResult, {
-					pathname: fileUrl.pathname.slice(config.root.pathname.length - 1),
+					pathname: '/@fs' + prependForwardSlash(fileUrl.pathname),
```

Comment:
Use same `pathname` pattern as standard `.astro` files. Without this change, markdown files might generate duplicate chunks with mismatched base paths